build break for samples/basic/minimal for board s32k148_evb

### Describe the bug

I am trying to build https://docs.zephyrproject.org/latest/samples/basic/minimal/README.html
especially Multithreading enabled, no preemption, timers disabled configuration.

i am getting below error
`01_zephyrproject/zephyr/soc/nxp/s32/s32k1/./soc.h:10:10: fatal error: fsl_port.h: No such file or directory
   10 | #include <fsl_port.h>`
getting this build error

### Regression

- [ ] This is a regression.

### Steps to reproduce

west build -b s32k148_evb -d build/s32k148_evb/mt-no-preempt-no-timers samples/basic/minimal -DCONF_FILE="common.conf mt.conf no-preempt.conf no-timers.conf arm.conf" 

### Relevant log output

```shell
-- west build: making build dir /localpath/zephyrproject/zephyr/build/s32k148_evb/mt-no-preempt-no-timers pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /localpath/zephyrproject/zephyr/samples/basic/minimal
-- CMake version: 3.22.1
-- Found Python3: /usr/bin/python3 (found suitable version "3.10.12", minimum required is "3.10") found components: Interpreter
-- Cache files will be written to: /home/localpath/.cache/zephyr
-- Zephyr version: 4.2.99 (/localpath/zephyrproject/zephyr)
-- Found west (found suitable version "1.4.0", minimum required is "0.14.0")
-- Board: s32k148_evb, qualifiers: s32k148
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.17.2 (/home/localpath/zephyr-sdk-0.17.2)
-- Found toolchain: zephyr 0.17.2 (/home/localpath/zephyr-sdk-0.17.2)
-- Found Dtc: /home/localpath/zephyr-sdk-0.17.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.7.0", minimum required is "1.4.6")
-- Found BOARD.dts: /localpath/zephyrproject/zephyr/boards/nxp/s32k148_evb/s32k148_evb.dts
-- Generated zephyr.dts: /localpath/zephyrproject/zephyr/build/s32k148_evb/mt-no-preempt-no-timers/zephyr/zephyr.dts
-- Generated pickled edt: /localpath/zephyrproject/zephyr/build/s32k148_evb/mt-no-preempt-no-timers/zephyr/edt.pickle
-- Generated devicetree_generated.h: /localpath/zephyrproject/zephyr/build/s32k148_evb/mt-no-preempt-no-timers/zephyr/include/generated/zephyr/devicetree_generated.h
Parsing /localpath/zephyrproject/zephyr/samples/basic/minimal/Kconfig
Loaded configuration '/localpath/zephyrproject/zephyr/boards/nxp/s32k148_evb/s32k148_evb_defconfig'
Merged configuration '/localpath/zephyrproject/zephyr/samples/basic/minimal/common.conf'
Merged configuration '/localpath/zephyrproject/zephyr/samples/basic/minimal/mt.conf'
Merged configuration '/localpath/zephyrproject/zephyr/samples/basic/minimal/no-preempt.conf'
Merged configuration '/localpath/zephyrproject/zephyr/samples/basic/minimal/no-timers.conf'
Merged configuration '/localpath/zephyrproject/zephyr/samples/basic/minimal/arm.conf'
Configuration saved to '/localpath/zephyrproject/zephyr/build/s32k148_evb/mt-no-preempt-no-timers/zephyr/.config'
Kconfig header saved to '/localpath/zephyrproject/zephyr/build/s32k148_evb/mt-no-preempt-no-timers/zephyr/include/generated/zephyr/autoconf.h'
-- Found GnuLd: /home/localpath/zephyr-sdk-0.17.2/arm-zephyr-eabi/arm-zephyr-eabi/bin/ld.bfd (found version "2.38")
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/localpath/zephyr-sdk-0.17.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Using ccache: /usr/bin/ccache
-- Found gen_kobject_list: /localpath/zephyrproject/zephyr/scripts/build/gen_kobject_list.py
-- Configuring done
-- Generating done
-- Build files have been written to: /localpath/zephyrproject/zephyr/build/s32k148_evb/mt-no-preempt-no-timers
-- west build: building application
[1/134] Preparing syscall dependency handling

[2/134] Generating include/generated/zephyr/version.h
-- Zephyr version: 4.2.99 (/localpath/zephyrproject/zephyr), build: v4.2.0-1067-ga8ecd9404454
[9/134] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
FAILED: zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
ccache /home/localpath/zephyr-sdk-0.17.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCPU_S32K148 -DFSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL=1 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DPICOLIBC_LONG_LONG_PRINTF_SCANF -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGc
In file included from /localpath/zephyrproject/zephyr/modules/cmsis_6/./cmsis_core_m.h:24,
                 from /localpath/zephyrproject/zephyr/modules/cmsis_6/./cmsis_core.h:10,
                 from /localpath/zephyrproject/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:24,
                 from /localpath/zephyrproject/zephyr/include/zephyr/arch/arm/asm_inline.h:18,
                 from /localpath/zephyrproject/zephyr/include/zephyr/arch/arm/arch.h:30,
                 from /localpath/zephyrproject/zephyr/include/zephyr/arch/cpu.h:19,
                 from /localpath/zephyrproject/zephyr/include/zephyr/kernel_includes.h:36,
                 from /localpath/zephyrproject/zephyr/include/zephyr/kernel.h:17,
                 from /localpath/zephyrproject/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from /localpath/zephyrproject/zephyr/arch/arm/core/offsets/offsets.c:9:
/localpath/zephyrproject/zephyr/soc/nxp/s32/s32k1/./soc.h:10:10: fatal error: fsl_port.h: No such file or directory
   10 | #include <fsl_port.h>
      |          ^~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /localpath/zephyrproject/zephyr/build/s32k148_evb/mt-no-preempt-no-timers
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

linux

### Additional Context

_No response_

--- a/modules/hal_nxp/s32/CMakeLists.txt
+++ b/modules/hal_nxp/s32/CMakeLists.txt
@@ -80,11 +80,9 @@ if(CONFIG_HAS_MCUX)
   # Load MCUX SDK NG code.
   include(../mcux/mcux-sdk-ng/basic.cmake)
 
-  if(${MCUX_DEVICE} MATCHES "S32K1")
-    set_variable_ifdef(CONFIG_PINCTRL CONFIG_MCUX_COMPONENT_driver.port)
-
+  if(CONFIG_SOC_SERIES_S32K1)
+    set(CONFIG_MCUX_COMPONENT_driver.port ON)
     set_variable_ifdef(CONFIG_CPU_HAS_NXP_SYSMPU CONFIG_MCUX_COMPONENT_driver.sysmpu)
-
     set_variable_ifdef(CONFIG_HAS_MCUX_CACHE CONFIG_MCUX_COMPONENT_driver.cache_lmem)
   endif()
 

